pipelines/main/misc/doctest.yml

```diff
@@ -11,27 +13,38 @@ steps:
           - JuliaCI/julia#v1:
               # Drop default "registries" directory, so it is not persisted from execution to execution
               persist_depot_dirs: packages,artifacts,compiled
-              version: '1.6'
+              version: '1'
           - staticfloat/sandbox#v1:
-              rootfs_url: https://github.com/JuliaCI/rootfs-images/releases/download/v4.8/package_linux.x86_64.tar.gz
-              rootfs_treehash: "2a058481b567f0e91b9aa3ce4ad4f09e6419355a"
+              rootfs_url: https://github.com/JuliaCI/rootfs-images/releases/download/v5.11/package_linux.x86_64.tar.gz
+              rootfs_treehash: "3c62a3dd4116a1b698c103f1e090b824e4cc9ff5"
               uid: 1000
               gid: 1000
               workspaces:
                 # Include `/cache/repos` so that our `git` version introspection works.
                 - "/cache/repos:/cache/repos"
         commands: |
-          echo "--- Build Julia from source"
-          make --output-sync -j 6
+          # Download pre-built julia, extract into `usr/`
+          set -x
+          buildkite-agent artifact download --step "build_x86_64-linux-gnu" 'julia-*-linux-x86_64.tar.gz' .
+          mkdir -p ./usr
+          tar -C ./usr --strip-components=1 -zxf julia-*-linux-x86_64.tar.gz
+          ln -s ./usr/bin/julia ./julia
 
           echo "--- Print Julia version info"
+          export JULIA_NUM_THREADS=1
           ./julia -e 'using InteractiveUtils; InteractiveUtils.versioninfo()'
 
-          echo "--- Build Julia docs"
-          make docs
+          echo "--- TESTING HACK TO OVERRIDE DOCUMENTER BRANCH AND DEPLOYDOCS INVOCATION"
+          TARBALL_NAME="$$(echo julia-*-linux-x86_64.tar.gz)"
+          TARBALL_NAME="$${TARBALL_NAME%-linux-x86_64.tar.gz}-htmldocs.tar.gz"
+          ./julia --project=doc -e 'using Pkg; pkg"add Documenter#mp/deploy-archive"'
+          sed -i.bak -e "s&deploydocs(&deploydocs(archive=\"$${TARBALL_NAME}\",&" doc/make.jl
 
-          echo "--- Run Julia doctests"
-          JULIA_NUM_THREADS=1 make -C doc doctest=true
+          echo "--- Run Julia doctests/build HTML docs"
+          make -C doc html doctest=true
```

Comment:
```suggestion
          make -C doc deploy doctest=true
```

I think?

Reply:
We also need to have `DOCUMENTER_KEY` set because of https://github.com/JuliaLang/julia/blob/b11ccae719c5826cc5c1629356c0b057e25b1b45/doc/make.jl#L340. I am not sure if e.g. an empty string would work though..

Reply:
Alright, if `DOCUMENTER_KEY` is set to an empty string, Documenter should be able to unpack it fine into an empty file. However, I am not sure what `ssh` will do if it tries to read an empty key file ([with this `.ssh/config`](https://github.com/JuliaDocs/Documenter.jl/blob/d17d1a763805605fd243de2a0fc55473d6cfa214/src/Documenter.jl#L735-L741)).

So if that doesn't work, what might work is patching `make.jl` further with

```julia
struct BuildBotConfig <: Documenter.DeployConfig end
# Add these two lines:
Documenter.authentication_method(::Documenter.DeployConfig) = Documenter.HTTPS
Documenter.authenticated_repo_url(::Documenter.DeployConfig) = "git@github.com:JuliaLang/docs.julialang.org.git"
```

to use [the `HTTPS` authentication path](https://github.com/JuliaDocs/Documenter.jl/blob/d17d1a763805605fd243de2a0fc55473d6cfa214/src/Documenter.jl#L760-L771).